compiler/resolution/ResolutionCandidate.cpp

```diff
@@ -282,6 +310,11 @@ bool ResolutionCandidate::computeSubstitutions(Expr* ctx) {
         computeSubstitutionForDefaultExpr(formal, ctx);
         nDefault++;
       }
+
+      // todo handle param formals too
```

Comment:
There are a bunch of TODOs around. Could you try making .future tests for most of them?